backend/tests/ingestion/test_client.py

```
@@ -0,0 +1,91 @@
+            pass
+        def json(self):
+            return {"success": True}
+
+    side_effects = [
+        httpx.HTTPStatusError("Error", request=AsyncMock(), response=MockErrorResponse()),
+        httpx.HTTPStatusError("Error", request=AsyncMock(), response=MockErrorResponse()),
+        MockSuccessResponse()
+    ]
+
+    with patch.object(client.client, "get", side_effect=side_effects) as mock_get:
+        with patch("litigation_api.ingestion.client.logger.error") as mock_logger:
```

Ruff (0.15.11)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)